identity_credential/src/credential/did_configuration_resource.rs

```diff
@@ -0,0 +1,302 @@
+  use crate::error::Result;
+  use crate::Error;
+  use identity_core::common::{Object, OrderedSet};
+  use identity_core::common::Timestamp;
+  use identity_core::common::Url;
+  use identity_core::convert::FromJson;
+  use identity_did::did::CoreDID;
+  use identity_did::did::CoreDIDUrl;
+  use identity_did::service::Service;
+  use serde_json::json;
+
+  #[test]
+  fn builder() {
+    let issuer = Issuer::Url(Url::parse("did:example:issuer").unwrap());
+    let credential: Credential = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .origin(Url::parse("http://www.example.com").unwrap())
+      .build()
+      .unwrap();
+    let _config: DIDConfigurationResource = DIDConfigurationResource::new(vec![credential]);
+  }
+
+  #[test]
+  fn builder_no_issuer() {
+    let credential: Result<Credential> = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .origin(Url::parse("http://www.example.com").unwrap())
+      .build();
+
+    assert!(matches!(credential, Err(Error::MissingIssuer)));
+  }
+
+  #[test]
+  fn builder_no_origin() {
+    let issuer = Issuer::Url(Url::parse("did:example:issuer").unwrap());
+    let credential: Result<Credential> = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .expiration_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .build();
+
+    assert!(matches!(credential, Err(Error::MissingOrigin)));
+  }
+
+  #[test]
+  fn builder_no_expiration_date() {
+    let issuer = Issuer::Url(Url::parse("did:example:issuer").unwrap());
+    let credential: Result<Credential> = DomainLinkageCredentialBuilder::new()
+      .issuance_date(Timestamp::now_utc())
+      .issuer(issuer)
+      .origin(Url::parse("http://www.example.com").unwrap())
+      .build();
+
+    assert!(matches!(credential, Err(Error::MissingExpirationDate)));
+  }
+
+  #[test]
+  fn from_json() {
+    const JSON1: &str = include_str!("../../tests/fixtures/well-known-configuration.json");
+    let _well_known: DIDConfigurationResource = DIDConfigurationResource::from_json(JSON1).unwrap();
+  }
+
+  #[test]
+  fn create_service_multiple_origins() {
+    let mut domains = OrderedSet::new();
+    domains.append(Url::parse("https://foo.example-1.com").unwrap());
+    domains.append(Url::parse("https://bar.example-2.com").unwrap());
+
+    let service: Service<CoreDID, Object> = DIDConfigurationResource::create_linked_domain_service(
+      CoreDIDUrl::parse("did:example:123#foo").unwrap(),
+      domains,
+    ).unwrap();
+
+    // Add a new Service.
+    let service_from_json: Service<CoreDID, Object> = Service::from_json_value(json!({
+        "id":"did:example:123#foo",
+        "type": "LinkedDomains",
+        "serviceEndpoint": {
+          "origins": ["https://foo.example-1.com", "https://bar.example-2.com"]
```

Comment:
Nit: Would be slightly better to define these urls once and then include them here to prevent copy-paste mistakes.